vclib-openid/src/commonMain/kotlin/at/asitplus/wallet/lib/oidvci/OAuth2AuthorizationServer.kt

```diff
@@ -26,5 +26,10 @@ interface OAuth2AuthorizationServer {
 
     // TODO How is this supposed to happen when using an external Authorization Server?
     suspend fun verifyAndRemoveClientNonce(nonce: String): Boolean
+
+    /**
+     * Provide necessary [OAuth2AuthorizationServerMetadata] JSON for a client to be able to authenticate
+     */
+    suspend fun provideMetadata(): String
```

Comment:
Why not `KmmResult<OAuth2AuthorizationServerMetadata>`?

Reply:
According to the spec it should return json encoded metadata. Wouldnt KmmResult interfere with the content-negotiation between client-server downstream?

Reply:
In a remote setting, the implementation would fetch the metadata from the remote service, decode it and pass it on (or an failure wrapped in a `KmmResult`).
Locally, passing the object makes even more sense.